### DISCUSSION TITLE
关于SQLite中字段无法便捷地设定外键（About the inability to set foreign keys conveniently for fields in SQLite）

### DISCUSSION THREAD

**Discussion by Gaven-F** [dbeaver/dbeaver#20415] [Q&A] [answered]
### Problem Description

I just downloaded DBeaver today to replace Navicat, and I feel satisfied with the initial use (I'm just a wasted college student who only knows a few simple operations). First of all, in some basic operations about Mysql, it perfectly replaces the operations I need on Navicat; but when I use it to operate SQLite, I found that it seems to be unable to set foreign keys on the GUI. 
To be more rigorous, I went to test on Navicat, which has a visual interface for setting foreign keys for SQLite, and similarly, on SQLite Studio. ### Expected response If I have a problem with the operation please leave a comment telling me the appropriate way to do it, or hint me how to do it on SQLite to set foreign keys (on the created table and when creating the table). Of course, the most expected result is to have visual interface for easy operation.

 ---

### 问题描述

我今天刚刚下载了DBeaver来替代Navicat，初步使用的情况下我觉得很满意（我只是一个虚度光阴的大学生，只会一些简单的操作）。
首先是在有关Mysql的一些基础操作上，它完美的替代了Navicat上我需要的操作；但是当我使用它操作SQLite的时候，我发现它似乎不能在GUI上进行外键的设置。

为了比较严谨，我去Navicat上进行测试，它对于SQLite设置外键有可视化界面，同样，在SQLite Studio上，也是有可视化界面。

### 期望的回复

如果是我操作有问题请留言告诉我相应的操作方法，或者提示我一下在SQLite上进行如何设置外键（在已创建的表上以及创建表时）。
当然，最期望得到的结果是有可视化界面方便操作。

---

## Attaching screenshots(附上截图)

- Navicat

![image](https://github.com/dbeaver/dbeaver/assets/133743603/b00f1dd4-5ce3-436f-a409-713e5a607b5b)

- SQLiteStudio

![image](https://github.com/dbeaver/dbeaver/assets/133743603/4e33dcf1-1183-4995-bb0d-11a2fe7b8365)

**Comment by E1izabeth** [comment, marked as answer]
You can create foreign keys.
![image](https://github.com/dbeaver/dbeaver/assets/28875055/9da57c0d-90f8-4065-a4d3-1a79166b5a5d)

![image](https://github.com/dbeaver/dbeaver/assets/28875055/e2ad47b5-56f3-4d62-a2a0-af51ea313092)

**Comment by Gaven-F** [comment]
Thanks for the reply, I think I'm a dummy.... 
I forgot to set it to advanced view, sorry
